src/Server/Services/Analytics/AnalyticsService.lua

```diff
@@ -0,0 +1,575 @@
+	GameAnalytics:setEnabledVerboseLog(false)
+	GameAnalytics:setEnabledDebugLog(false)
+	
+	GameAnalytics:setEnabledAutomaticSendBusinessEvents(false)
+	GameAnalytics:setEnabledReportErrors(false)
+	
+	GameAnalytics:initServer(GAME_KEY, SECRET_KEY)
+end
+
+function AnalyticsService:KnitStart()
+	self.Client.LogEvent:Connect(function(player: Player, data: {
+		event: string,
+		value: number?
+	})
+		self:LogPlayerEvent({
+			userId = player.UserId,
+			event = data.event,
+			value = data.value
+		})
+	end)
+	
+	MarketplaceService.PromptBundlePurchaseFinished:Connect(
+		function(player: Player, bundleId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Bundle",
+				id = bundleId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	MarketplaceService.PromptGamePassPurchaseFinished:Connect(
+		function(player: Player, gamePassId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "GamePass",
+				id = gamePassId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	if LOG_DEV_PRODUCT_PURCHASES then
+		MarketplaceService.PromptProductPurchaseFinished:Connect(
+			function(player: Player, productId: number, purchased: boolean)
+				if not purchased then
+					return
+				end
+				
+				self:LogMarketplacePurchase({
+					userId = player.UserId,
+					itemType = "Product",
+					id = productId,
+					cartType = "PromptPurchase",
+				})
+			end
+		)
+	end
+	
+	MarketplaceService.PromptPurchaseFinished:Connect(
+		function(player: Player, assetId: number, purchased: boolean)
+			if not purchased then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Asset",
+				id = assetId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+	
+	MarketplaceService.PromptSubscriptionPurchaseFinished:Connect(
+		function(player: Player, subscriptionId: string, didTryPurchasing: boolean)
+			if not didTryPurchasing then
+				return
+			end
+			
+			self:LogMarketplacePurchase({
+				userId = player.UserId,
+				itemType = "Subscription",
+				id = subscriptionId,
+				cartType = "PromptPurchase",
+			})
+		end
+	)
+end
+
+--[=[
+	Used to track player events (example: player killed an enemy, player completed a mission, etc.)
+	
+	Examples
+	```lua
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:KilledEnemy",
+		value = 1 -- Killed 1 enemy
+	})
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:CompletedMission",
+		value = 1 -- Completed 1 mission
+	})
+	AnalyticsService:LogPlayerEvent({
+		userId = player.UserId,
+		event = "Player:Death",
+		value = 1
+	})
+	```
+]=]
+function AnalyticsService:LogPlayerEvent(data: PlayerEvent)
+	assert(typeof(data) == "table", "AnalyticsService.LogPlayerEvent - data is required")
```

Comment:
Using assert will halt execution of the function that called this function, and therefore error-handling would have to be implemented for each function that calls LogPlayerEvent (Same applies for other functions in this service that use similar behavior with assert).

I would recommend implementing a promise - return the promise from this function. The promise should reject if a condition isn't met (The ones defined by assert), and should resolve if all conditions are met.

With Moonwave, you can define a custom one-off type by doing VARIABLE<T> (Eg. Promise<T>).

This could look something like:

```
function AnalyticsService:LogPlayerEvent(data: PlayerEvent): {[any]: any}

return Promise.new(function(Resolve, Reject)

if not condition then

return Reject("Condition not met!")

end

return Resolve("Condition met!")

end)

end
```